bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -5376,14 +5374,21 @@ public Image getImageById(long id) throws TskCoreException {
 					}
 				}
 				long size = rs1.getLong("size"); //NON-NLS
-				return new Image(this, obj_id, type, ssize, name,
+				
+				s3 = connection.createStatement();
```

Comment:
I believe you can do this in just one round trip top the case database by doing a JOIN with the data_source_info table, instead of two separate queries.